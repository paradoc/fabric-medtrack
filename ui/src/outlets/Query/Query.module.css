.container {
  display: grid;
  grid-template-rows: min-content max-content;
  row-gap: 36px;
}

.inputField {
  display: grid;
  grid-template-columns: 100px 260px max-content;
  column-gap: 12px;
  align-items: center;
}

.inputField > input {
  height: 32px;
  border-radius: 4px;
  border: 1px solid lightgray;
}

.inputField > label {
  font-weight: bold;
  color: #333;
}

.inputField > span {
  font-size: 14px;
  color: #eb455f;
}

.submit {
  border: none;
  height: 48px;
  width: 200px;
  font-weight: bold;
  color: #555;
  border-radius: 4px;
}