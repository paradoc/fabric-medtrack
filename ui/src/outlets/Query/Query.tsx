import React, { useState } from 'react'
import { useForm } from 'react-hook-form'
import { Navigate } from 'react-router'

import styles from './Query.module.css'

type QueryFormData = {
  generic: string
  start: string
  end: string
}

export default function Query() {
  const [query, setQuery] = useState<string>()
  const {
    register,
    handleSubmit,
    reset,
    formState: { errors },
  } = useForm<QueryFormData>()

  const onSubmit = handleSubmit(async ({ generic, start, end }) => {
    setQuery(`./query?g=${generic}&s=${start}&e=${end}`)
    reset()
  })

  return query ? (
    <Navigate to={query} relative="path" />
  ) : (
    <form className={styles.container} onSubmit={onSubmit}>
      <div className={styles.inputField}>
        <label htmlFor="generic">Generic</label>
        <input
          id="generic"
          type="text"
          {...register('generic', { required: true })}
          placeholder="comma delimited (ie. med1,med2)"
          data-testid="generic"
        />
        <span>{errors.generic && '* required'}</span>
      </div>

      <div className={styles.inputField}>
        <label htmlFor="start">Start</label>
        <input
          id="start"
          type="date"
          {...register('start', { required: true })}
          data-testid="start"
        />
        <span>{errors.start && '* required'}</span>
      </div>

      <div className={styles.inputField}>
        <label htmlFor="end">End</label>
        <input
          id="end"
          type="date"
          {...register('end', { required: true })}
          data-testid="end"
        />
        <span>{errors.end && '* required'}</span>
      </div>

      <input
        type="submit"
        className={styles.submit}
        value="Query"
        data-testid="submit"
      />
    </form>
  )
}
