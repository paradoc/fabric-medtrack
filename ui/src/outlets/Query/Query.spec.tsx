import React from 'react'
import { act, fireEvent, render, waitFor } from '@testing-library/react'
import Query from './Query'
import { MemoryRouter } from 'react-router'

describe('Query', () => {
  test('should match snapshot', () => {
    const { container } = render(<Query />)
    expect(container).toMatchSnapshot()
  })

  test('should check behaviors', async () => {
    const { container, getByTestId } = render(
      <MemoryRouter>
        <Query />
      </MemoryRouter>
    )
    const submit = getByTestId('submit')
    const generic = getByTestId('generic')
    const start = getByTestId('start')
    const end = getByTestId('end')
    act(() => {
      fireEvent.change(generic, { target: { value: 'test-generic' } })
      fireEvent.change(start, { target: { value: '2022-12-01' } })
      fireEvent.change(end, { target: { value: '2022-12-02' } })
      fireEvent.click(submit)
    })
    expect(container).toMatchSnapshot()
  })
})
