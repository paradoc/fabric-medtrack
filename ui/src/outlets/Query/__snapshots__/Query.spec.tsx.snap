// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Query should check behaviors 1`] = `
<div>
  <form
    class="container"
  >
    <div
      class="inputField"
    >
      <label
        for="generic"
      >
        Generic
      </label>
      <input
        data-testid="generic"
        id="generic"
        name="generic"
        placeholder="comma delimited (ie. med1,med2)"
        type="text"
      />
      <span />
    </div>
    <div
      class="inputField"
    >
      <label
        for="start"
      >
        Start
      </label>
      <input
        data-testid="start"
        id="start"
        name="start"
        type="date"
      />
      <span />
    </div>
    <div
      class="inputField"
    >
      <label
        for="end"
      >
        End
      </label>
      <input
        data-testid="end"
        id="end"
        name="end"
        type="date"
      />
      <span />
    </div>
    <input
      class="submit"
      data-testid="submit"
      type="submit"
      value="Query"
    />
  </form>
</div>
`;

exports[`Query should match snapshot 1`] = `
<div>
  <form
    class="container"
  >
    <div
      class="inputField"
    >
      <label
        for="generic"
      >
        Generic
      </label>
      <input
        data-testid="generic"
        id="generic"
        name="generic"
        placeholder="comma delimited (ie. med1,med2)"
        type="text"
      />
      <span />
    </div>
    <div
      class="inputField"
    >
      <label
        for="start"
      >
        Start
      </label>
      <input
        data-testid="start"
        id="start"
        name="start"
        type="date"
      />
      <span />
    </div>
    <div
      class="inputField"
    >
      <label
        for="end"
      >
        End
      </label>
      <input
        data-testid="end"
        id="end"
        name="end"
        type="date"
      />
      <span />
    </div>
    <input
      class="submit"
      data-testid="submit"
      type="submit"
      value="Query"
    />
  </form>
</div>
`;
